### PR TITLE
Different column name renaming scheme

### PR DESCRIPTION
**Pull request by rozap**
* Case sensitive collisions could happen, now they
  shouldn't happen...
* Added tests, fixed other tests

### CODE REVIEW

**Comment by rozap** [comment]
Note that this is for EN-6393

File: es6-lib/soql/soql.js [hunk 1]
```diff
@@ -14,9 +14,27 @@ class SoQL {
    * TODO: this is necessary, but is this sufficient?
```
Comment (kaidamasaki) [inline es6-lib/soql/soql.js]: I love this TODO.

File: es6-lib/soql/soql.js [hunk 2]
```diff
@@ -14,9 +14,27 @@ class SoQL {
...
    */
   _launderName(name) {
+    name = name.trim();
+    var start = name[0];
+    var isNumber = !_.isNaN(parseInt(start));
```
Comment (kaidamasaki) [inline es6-lib/soql/soql.js]: Consider `_.isNumber` or at the very least `_.parseInt(start)`.

Comment (kaidamasaki) [inline es6-lib/soql/soql.js]: Oh..  You're using underscore.  You should probably switch to lodash.

File: es6-lib/soql/soql.js [hunk 3]
```diff
@@ -14,9 +14,27 @@ class SoQL {
...
+    var isUpper = start.toUpperCase() === start;
+
+
+    // Suppose we have columns in one layer
+    // called `NAME` and `name`, snakecase itself will convert
+    // these to `name` and `name`. So we need to convert them
```
Comment (kaidamasaki) [inline es6-lib/soql/soql.js]: What happens if you have `NAME`, `Name` and `name`?

Comment (rozap) [inline es6-lib/soql/soql.js]: i'm punting on that for now - i'd want to think a little bit more on where that logic would go. this is enough to unblock the people who have duplicate columns due to case

**Pull request merged**